### PR TITLE
Add types

### PR DESCRIPTION
**Pull request by harish-sethuraman**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [x] **metadata update**

### Motivation / Use-Case
Adds type definitions in readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -637,10 +649,17 @@ export default CustomSquare;
 
 ### `styleTagTransform`
 
-Type: `String | Function`
+Type:
+
+```ts
+type styleTagTransform =
+  | string
+  | ((css: string, styleElement: HTMLStyleElement, options: any) => void);
```
Comment (alexander-akait) [inline README.md:657]: Why `any`?

Comment (harish-sethuraman) [reply README.md:657]: These options are injected from JS like  
```js
 style.use({
  insertInto: document.body,
  additionalStyles: '.some-element {color: red;}'
});
```
this allows to set any options?

Comment (alexander-akait) [reply README.md:657]: Let use `Record<string, any>`, because other types are invalid

Comment (harish-sethuraman) [reply README.md]: Done

**Review by alexander-akait — APPROVED** [review]

**Comment by harish-sethuraman** [comment]
Have to rename `Change` -> `change` in commit message 😑

**Comment by alexander-akait** [comment]
I will fix :+1:

**Comment by harish-sethuraman** [comment]
Thanks

**Pull request merged**